src/main/java/com/giantshoulder/pageobject/MainPage.java

```diff
@@ -0,0 +1,131 @@
+package com.giantshoulder.pageobject;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.FindBy;
+
+import static com.giantshoulder.util.WebElementHandler.*;
+
+/**
+ * @author Gabriella_Havadtoi
+ */
+public class MainPage extends BasePage {
+
+    public MainPage(ChromeDriver driver) {
+        super(driver);
+    }
+
+    @FindBy(id = "element3421730-content")
+    WebElement subscribeButton;
+
+    @FindBy(css = ".fc-button.fc-cta-consent.fc-primary-button")
+    WebElement dataConsentButton;
+
+    @FindBy(id = "element1294352-content")
+    WebElement dailyInspirationBoxTitle;
+
+    @FindBy(id = "element1294352-content")
+    WebElement learnFasterNewsletterTypeButton;
+
+    @FindBy(id = "element1294351-content")
+    WebElement stopProcrastinatingNewsletterTypeButton;
+
+    @FindBy(id = "element1294355-content")
+    WebElement improveFocusNewsletterTypeButton;
+
+    @FindBy(id = "element1294354-content")
+    WebElement reachYourGoalsNewsletterTypeButton;
+
+    @FindBy(css = "#element1294344_form [data-field='email'] input")
+    WebElement emailInputFieldForLearnFaster;
+
+    @FindBy(css = "#element1294347_form [type='email']")
+    WebElement emailInputFieldForStopProcrastinating;
+
+    @FindBy(css = "#element1294338_form [type='email']")
+    WebElement emailInputFieldForReachYourGoals;
+
+    @FindBy(css = "#element1294341_form [type='email']")
+    WebElement emailInputFieldForImproveFocus;
+
+    @FindBy(css = "button[data-element-id='1294344']")
+    WebElement submitEmailButtonForLearnFaster;
+
+    @FindBy(css = "button[data-element-id='1294347']")
+    WebElement submitEmailButtonForStopProcrastinating;
+
+    @FindBy(css = "button[data-element-id='1294338']")
+    WebElement submitEmailButtonForReachYourGoals;
+
+    @FindBy(css = "button[data-element-id='1294341']")
+    WebElement submitEmailButtonForImproveFocus;
+
+    @FindBy(css = "img[src*='big-success']")
+    WebElement emailValidatedPopup;
+
+    public void clickOnSubscribeButton() {
+        waitForElementToBeVisible("subscribe button", driver, subscribeButton);
+        clickOn("subscribe button", driver, subscribeButton);
+    }
+
+    public void acceptDataConsentPanel() {
+        waitForElementToBeVisible("consent button", driver, dataConsentButton);
+        clickOn("data consent button", driver, dataConsentButton);
+    }
+
+    public void clickOnNewsLetterButtonFor(String newsLetterType) {
+        waitForElementToBeVisible("subscribe panel", driver, dailyInspirationBoxTitle);
+        switch (newsLetterType.toLowerCase()) {
```

Comment:
Using switch like this will become confusing and hard to debug very fast.

Reply:
Fixed in 5e8e0dc07a1b347edd3a35b7fc060122e44581be!